### PR TITLE
improve OS_RELEASE_ID definition

### PR DESCRIPTION
**Pull request by kewlfft**
adopt the [proposal](https://github.com/gentoo/portage/pull/349#issuecomment-425483117) fixing `OS_RELEASE_ID`

### CODE REVIEW

**Comment by kewlfft** [comment]
It is fixing the case for example: 
ID=arch\"007\"
The actual ID is arch"007" and not arch\007\

**Comment by zmedico** [comment]
According to https://www.freedesktop.org/software/systemd/man/os-release.html, it's supposed to contain "shell-compatible variable assignments", so using `source` appears to be the right thing to do.

File: bin/etc-update [hunk 1]
```diff
@@ -32,7 +32,7 @@ get_config() {
 		"${PORTAGE_CONFIGROOT}"etc/etc-update.conf)
 }
 
-OS_RELEASE_ID=$(cat /etc/os-release 2>/dev/null | grep '^ID=' | cut -d'=' -f2 | sed -e 's/"//g')
+OS_RELEASE_ID=$(source /etc/os-release >/dev/null 2>&1; echo $ID)
```
Comment (zmedico) [inline bin/etc-update]: Just in case $ID contains special shell characters, we should use `echo "$ID"`.

Comment (kewlfft) [reply bin/etc-update]: Ok then the `OS_RELEASE_ID` variable is not really necessary and we could simply use `ID`

Comment (zmedico) [reply bin/etc-update]: Well, the isolation provided by the subshell is useful, since lots of things could go wrong when a file is sourced.

Comment (kewlfft) [reply bin/etc-update]: yes sure, agreed.

Comment (mgorny) [reply bin/etc-update]: Also `${ID}`, please, for consistency with what we tend to use everywhere.

**Comment by kewlfft** [comment]
Signed-off-by: Xrjy SSG xrjy@nygb.rh.bet (rot13)

File: bin/etc-update (hunk 1 above)
Comment (eli-schwartz) [reply bin/etc-update]: When echoing a variable, shell characters in the variable definition shouldn't be reinterpreted? Quotes should only really matter for word splitting, which as a matter of good practice should still be quoted on those grounds alone.

The actual practical effect would be that 
```
$ ID="foo    bar"
$ echo $ID
foo bar
$ echo "$ID"
foo    bar
$ IFS=f
$ echo $ID
 oo    bar
$ echo "$ID"
foo    bar
```
Failure to quote, means the shell collapses whitespace (usually) due to reinterpreting it by way of `$IFS`.

**Comment by kewlfft** [comment]
Can I close this PR as it is included in #372 which is being discussed ?

**Comment by zmedico** [comment]
> Can I close this PR as it is included in #372 which is being discussed ?

Yes, please do.

**Pull request closed without merging**